src/Core/Managed/Shared/Extensibility/Implementation/Property.cs

```diff
@@ -120,6 +150,18 @@ public static string SanitizeAvailabilityMessage(this string value)
             return TrimAndTruncate(value, Property.MaxAvailabilityMessageLength);
         }
 
+        public static void SanitizeTelemetryContext(this TelemetryContext telemetryContext)
+        {
+            SanitizeComponentContext(telemetryContext.Component);            
```

Comment:
typical event doesn't have all the context properties set. Also every get/set operation is a lookup and copying of the dictionary. 

Instead of this implementation - loop thru the tags collection directly and only truncate if needed.